pkg/ccip/chainaccessor/event.go

```diff
@@ -0,0 +1,42 @@
+package chainaccessor
+
+import (
+	"errors"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/chainaccessor"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ton/pkg/ccip/bindings/onramp"
+)
+
+// convertCCIPMessageSent converts a TON CCIPMessageSent event to a generic chainaccessor.SendRequestedEvent event
+func (a *TONAccessor) convertCCIPMessageSent(
```

Comment:
This is a more general structure issue but we should have this relationship between these types expressed with an interface and types that implement (e.g., `TypeMapper[A, B]`), and we can define a set of these mappers we manage vs. random functions that do the mapping.